### PR TITLE
Feature/update error response handling

### PR DESCRIPTION
**Pull request by ts-nitish-a-jain**
# Description
This PR include:
- Modifications for handling 401/403 error responses (which are in a different format than the rest) as received from the backend.
- Minor refactor and fix for detekt warnings

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

**Comment by ts-nitish-a-jain** [comment]
Local shows me 4% ↑ image. What am I doing differently?

image

File: miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt [hunk 1]
```diff
@@ -63,52 +64,79 @@ internal class ApiClient @VisibleForTesting constructor(
...
+    ).message()
 }
 
-internal data class ErrorResponse(
+internal data class HttpErrorResponse(
     val code: Int,
+    override val message: String
+) : ErrorResponse
+
+internal data class ApiCatalogErrorResponse(
```
Comment (corycaywood) [inline miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt]: I think this name is exposing details about the API that we don't need to know as the SDK. How about `AuthErrorResponse` instead?

Comment (Climbatize) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt]: On iOS we called it UnauthorizedData, as opposed to ErrorData we already had in place

Comment (ts-nitish-a-jain) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt]: The choice I made was by considering future reference and readability. I would definitely forget why there are two types from the backend. The SDK is managing the deviation of response types from the backend. So, it is aware indeed.

I had two options, mark that domain-effect into the names of behavior or record it as a comment. I choose the former.

Unless, there is a strong opinion on renaming it, I would like to keep it the same.

Comment (corycaywood) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt]: I think something that refers to it being related to Authorization is more explanatory. It's not the SDK's responsibility to document why certain choices were made by the API side. We just have to document that there is a different response for different response codes.

Also, keep in mind that this is a public SDK, so outside developers will have no idea what API Catalog is.

Comment (ts-nitish-a-jain) [reply miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt]: Thanks @corycaywood, I think that's reasonable. 👍

**Comment by corycaywood** [comment]
> Local shows me 4% ↑ image. What am I doing differently?
> 
> image

CodeCov works in mysterious ways. Sometimes it will show a different percentage when there were only changes to build files and not to source files.

**Comment by ts-nitish-a-jain** [comment]
> > Local shows me 4% ↑ image. What am I doing differently?
> > image
> 
> CodeCov works in mysterious ways. Sometimes it will show a different percentage when there were only changes to build files and not to source files.

That's some interesting stuff..! 🤣

**Pull request merged**